splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/SpliceBaseOperation.java

```diff
@@ -428,10 +459,24 @@ public void reopenCore() throws StandardException{
     public ExecRow getNextRowCore() throws StandardException{
         try{
             if(execRowIterator.hasNext()){
-                locatedRow= execRowIterator.next();
-                if(LOG.isTraceEnabled())
-                    SpliceLogUtils.trace(LOG,"getNextRowCore %s locatedRow=%s",this,locatedRow);
-                return locatedRow;
+                try {
+                    locatedRow = execRowIterator.next();
+                    if (LOG.isTraceEnabled())
+                        SpliceLogUtils.trace(LOG, "getNextRowCore %s locatedRow=%s", this, locatedRow);
+                    ExecRow result = locatedRow;
+                    returnedRows = true;
+                    return result;
+                } catch (ResubmitDistributedException e) {
+                    if (!returnedRows) {
+                        LOG.warn("The query consumed too many resources running in control mode, resubmitting in Spark");
```

Comment:
Is this case possible: query returned few rows from control and then it resubmitted the job to spark, which got all of the rows, but that would make more rows than correct number of rows.